css/geometry-1/structured-serialization.html

```diff
@@ -0,0 +1,136 @@
+<!DOCTYPE html>
+<title>Geometry Interfaces: Serialize and deserialize</title>
+<link rel="help" href="https://drafts.fxtf.org/geometry/#structured-serialization">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+function clone(obj) {
+  return new Promise((resolve, reject) => {
+    const channel = new MessageChannel();
+    channel.port2.onmessage = e => resolve(e.data);
+    channel.port1.postMessage(obj);
+  });
+}
+function defineThrowingGetters(object, attrs, message) {
+  for (let attr of attrs) {
+    Object.defineProperty(object, attr, {
+      get: () => assert_unreached(`getter for ${attr} ${message}`)
+    });
+  }
+}
+
+[
+  ["DOMPointReadOnly", "x", "y", "z", "w"],
+  ["DOMPoint", "x", "y", "z", "w"],
+  ["DOMRectReadOnly", "x", "y", "width", "height"],
+  ["DOMRect", "x", "y", "width", "height"],
+  ["DOMQuad", "p1", "p2", "p3", "p4"],
+  ["DOMMatrixReadOnly", "a", "b", "c", "d", "e", "f", "m11", "m12", "m13", "m14", "m21", "m22", "m23", "m24", "m31", "m32", "m33", "m34", "m41", "m42", "m43", "m44", "is2D"],
+  ["DOMMatrix", "a", "b", "c", "d", "e", "f", "m11", "m12", "m13", "m14", "m21", "m22", "m23", "m24", "m31", "m32", "m33", "m34", "m41", "m42", "m43", "m44", "is2D"],
+].forEach(([constr, ...attrs]) => {
+  const prefix = `${constr} clone:`;
+
+  promise_test(t => {
+    const object = new self[constr]();
+    return clone(object).then(other => {
+      assert_not_equals(object, other);
+      assert_class_string(other, constr);
+      for (let attr of attrs) {
+        if (constr === "DOMQuad") {
+          assert_not_equals(other[attr], object[attr], attr);
+          assert_class_string(other[attr], "DOMPoint", attr);
+          assert_equals(other[attr].x, object.p1.x, `${attr}.x`);
+          assert_equals(other[attr].y, object.p1.y, `${attr}.y`);
+          assert_equals(other[attr].z, object.p1.z, `${attr}.z`);
+          assert_equals(other[attr].w, object.p1.w, `${attr}.w`);
+        } else {
+          assert_equals(other[attr], object[attr], attr);
+        }
+      }
+      t.done();
+    });
+  }, `${prefix} basic`);
+
+  promise_test(t => {
+    const object = new self[constr]();
+    object.foo = "bar";
+    return clone(object).then(other => {
+      assert_false("foo" in other, 'foo should not exist in other');
+      t.done();
+    });
+  }, `${prefix} custom property`);
+
+  promise_test(t => {
+    const object = new self[constr]();
+    defineThrowingGetters(object, attrs, 'should not be copied');
```

Comment:
I mean that the custom getter should not be copied to *other*. But I guess this is more confusing than helpful, so let's remove it, or move to a JS comment...